ts/src/test/scala/ch/epfl/ts/test/data/StrategyParameters.scala

```diff
@@ -0,0 +1,105 @@
+package ch.epfl.ts.test.data
+
+import org.junit.runner.RunWith
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.FunSuite
+import ch.epfl.ts.data.Currency
+import scala.util.Try
+import ch.epfl.ts.data.CurrencyPairParameter
+import ch.epfl.ts.data.StrategyParameters
+import ch.epfl.ts.data.CoefficientParameter
+import ch.epfl.ts.data.Parameter
+import ch.epfl.ts.data.ParameterTrait
+
+@RunWith(classOf[JUnitRunner])
+class StrategyParametersTests extends FunSuite {
+  /**
+   * Common test values
+   */
+	val currencies1 = (Currency.EUR, Currency.CHF)
+	val currencies2 = (Currency.GBP, Currency.USD)
+  val coefficient1 = 0.2342341341
+  val coefficient2 = -0.2342341341
+  
+  private type CurrencyPair = (Currency.Currency, Currency.Currency)
+  
+  /** Generic function to test basic functionality expected from Parameter subclasses */
+  def testConcreteParameter[T](parameterTrait: ParameterTrait[T], validValues: Iterable[T], invalidValues: Iterable[T]) = {
+    test(parameterTrait + " should give back the value it was instantiated with") {
+      for {
+        myValue <- validValues
+        parameter = parameterTrait.getInstance(myValue)        
+      } yield assert(parameter.get() == myValue, "Should give back value " + myValue)
+    }
+    
+    test(parameterTrait + " should reject invalid values") {
+      for {
+        myValue <- invalidValues
```

Comment:
How are you going to define invalidValues? This sounds a bit crazy to me. I would just test some concrete examples (e.g. parameter is a numerical value => max(validValues) + 1000 is an invalid value).